src/sfizz/MidiState.h

```diff
@@ -133,9 +134,51 @@ class MidiState
      */
     void resetAllControllers(int delay) noexcept;
 
+    /**
+     * @brief Notify all the controllers to the observer
+     */
+    void notifyAllControllers() noexcept;
+
     const EventVector& getCCEvents(int ccIdx) const noexcept;
     const EventVector& getPitchEvents() const noexcept;
 
+    /**
+     * @brief Observer of controller change events
+     */
+    class ControllerChangeObserver {
+    public:
+        virtual ~ControllerChangeObserver() {}
+        virtual void onAllControllersReset() noexcept = 0;
+        virtual void onControllerChange(int ccNumber, float ccValue) noexcept = 0;
+    };
+
+    /**
+     * @brief Observer of controller change events, which records the
+     * information in a linked list of non-duplicated controller changes.
+     */
+    class ControllerChangeRecorder : public ControllerChangeObserver {
```

Comment:
Should this class be within the MidiState one?

Reply:
How you prefer.. I certainly don't have issue with the way how it is.